src/Chainweb/Pact/Backend/Types.hs

```diff
@@ -297,6 +297,9 @@ data Checkpointer logger = Checkpointer
       -- ^ commits pending modifications to block, with the given blockhash
     , _cpDiscard :: !(IO ())
       -- ^ discard pending block changes
+    , _cpGetEarliestBlock :: !(IO (Maybe (BlockHeight, BlockHash)))
```

Comment:
Why is the the `Maybe` needed? Is it possible to have a chainweb-node without any block in the database?

Reply:
> Why is the the `Maybe` needed? Is it possible to have a chainweb-node without any block in the database?

I think it shouldn't be. We should probably throw an exception in that case.